Tools/autotest/common.py

```diff
@@ -2601,6 +2651,111 @@ def test_arm_feature(self):
         if self.armed():
             if not self.disarm_vehicle():
                 raise NotAchievedException("Failed to DISARM")
+
+        # altitude source consistency checks - ahrs and GPS:
+        if not self.is_rover():
+            self.context_push()
+            ex = None
+            try:
+                # n.b. no reason we shouldn't run this for Rover, I just
+                # can't get the altitude to drift :-)
+                self.start_subtest("Ensure we can't arm if AHRS and GPS don't agree on height")
+                if self.is_copter():
+                    # change to a mode which actually cares about height:
+                    self.change_mode("LOITER")
+                self.progress("First ensure we can arm")
+                self.wait_ready_to_arm()
+                self.arm_vehicle()
+                self.disarm_vehicle()
+                self.progress("Hurt the baro and try again")
+                vfr_hud_init = self.mav.recv_match(type="VFR_HUD", blocking=True, timeout=1)
```

Comment:
That's a short timeout, doubtful that will work on Azure - we already have issues with other messages not being received in time.